how-to/postgrest.md

```diff
@@ -397,4 +397,162 @@ Now, you can perform a keyword-based search directly from your web browser. Simp
+
+INSERT INTO terms (term, readings) VALUES ('牛乳', ARRAY['ギュウニュウ', 'ミルク']);
+INSERT INTO terms (term, readings) VALUES ('PostgreSQL', ARRAY['ポスグレ', 'posugure', 'postgres']);
+INSERT INTO terms (term, readings) VALUES ('Groonga', ARRAY['guru-nga', 'グルンガ','ぐるんが','gurunga']);
+INSERT INTO terms (term, readings) VALUES ('PGroonga', ARRAY['pi-ji-runga', 'ピージールンガ','ぴーじーるんが','pg','ピージー']);
+```
+
+### Set Up PostgREST Permission
+
+```sql
+GRANT SELECT ON terms TO web_user;
+```
+
+### Create Auto Complete End Point
+
+```sql
+CREATE OR REPLACE FUNCTION autocomplete(keyword text) RETURNS SETOF text AS $$
+DECLARE
+  result text[];
+BEGIN
+  IF keyword = '' THEN
+    RETURN QUERY SELECT unnest(result);
+  ELSE
+    RETURN QUERY SELECT term FROM terms WHERE readings &^~ keyword;
+  END IF;
+END;
+$$ LANGUAGE plpgsql;
+```
+
+### Create a HTML with JavaScript
+
+Create following HTML file:
+
+```sh
+vi index.html
+```
+
+```html
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>PGroonga Auto Complete Search</title>
+    <link rel="stylesheet"
+          href="https://cdn.jsdelivr.net/npm/@tarekraafat/autocomplete.js@10.2.7/dist/css/autoComplete.min.css">
+    <style>
+        .center-container {
+            display: flex;
+            justify-content: center;
+            align-items: center;
+            min-height: 100vh;
+        }
+    </style>
+  </head>
+  <body>
+    <div class="center-container">
+        <form name="search" id="searchForm">
+            <input type="search" size="60" maxlength="60" name="key" id="autoComplete">
+        </form>
+
+        <script src="https://cdn.jsdelivr.net/npm/@tarekraafat/autocomplete.js@10.2.7/dist/autoComplete.min.js">
+        </script>
+        <script type="text/javascript">
+            const searchForm = document.getElementById('searchForm');
+
+            searchForm.addEventListener('submit', async (event) => {
+                event.preventDefault();
+                const query = document.getElementById('autoComplete').value;
+                
+                if (query.length === 0) {
+                    return;
+                }
+
+                const params = new URLSearchParams({keywords: query});
+                const response = await fetch(`http://localhost:3000/rpc/find_title?${params.ToString()}`);
+                const result = await response.json();
+
+                const preElement = document.getElementById('output');
+                if (result.length > 0) {
+                    const formattedValue = JSON.stringify(result, null, 2);
+                    preElement.textContent = formattedValue;
+                } else {
+                    preElement.textContent = 'No results found';
+                }
+            });
+
+            const dataSource = async (query) => {
+                const source = await fetch(`http://localhost:3000/rpc/autocomplete?keyword=${query}`);
```

Comment:
Could you use `URLSearchParams` here too?

Reply:
I just figure out that using `URLSearchParams` will break not only auto complete feature but search function too.

Reply:
My bad. I misunderstood how to use `URLSearchParams`.